### PR TITLE
fix: [#4470] Migrate off @azure/ms-rest-js

### PR DESCRIPTION
**Pull request by JhontSouth**
#minor

## Description
This PR replaces the uses of **_@azure/ms-rest-js_** in the following botbuilder-js libraries.

- [x] botbuilder
- [ ] botbuilder-ai
- [x] botframework-connector
- [ ] tool

## Specific Changes
  - Replaced the **_@azure/ms-rest-js_** imports with **_@azure/core-http_**.
  - Added **_@azure/core-http_** to libraries package.json.
  - Added **_@azure/core-util_** version 1.5.0 to resolutions.
  - Updated unit tests to work with core-http **_RestError_**.

## Testing
The following image shows some samples working after the update.
image
image

### CODE REVIEW

File: package.json
Comment (ceciliaavila) [inline package.json:1]: I think this change should be in the last PR to avoid the checks to fail.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: package.json
Comment (JhontSouth) [reply package.json:1]: this is the main PR, I'll create the other PR's pointing to this one

Comment (ceciliaavila) [reply package.json:1]: Mmm...ok, then you should check why the pipelines are failing.

File: libraries/botframework-connector/src/connectorApi/connectorClient.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 import * as Mappers from './models/mappers';
 import * as Models from './models';
-import * as msRest from '@azure/ms-rest-js';
+import { ServiceClientCredentials } from "@azure/core-http"
```
Comment (ceciliaavila) [inline libraries/botframework-connector/src/connectorApi/connectorClient.ts:8]: Use single quotes for the import.

File: libraries/botframework-connector/src/connectorApi/operations/conversations.ts [hunk 2]
```diff
@@ -723,13 +723,13 @@ const getConversationMembersOperationSpec: msRest.OperationSpec = {
...
-            element: {
-                type: {
-                    name: "Composite",
-                    className: "ChannelAccount"
-                }
+          name: "Sequence",
+          element: {
+            type: {
+              name: "Composite",
+              className: "ChannelAccount"
```
Comment (ceciliaavila) [inline libraries/botframework-connector/src/connectorApi/operations/conversations.ts:730]: What happened with the indentation? Is the new one correct?

File: libraries/botframework-connector/src/tokenApi/operations/userToken.ts [hunk 3]
```diff
@@ -3,7 +3,7 @@
  * Licensed under the MIT License.
  */
 
-import * as msRest from '@azure/ms-rest-js';
+import { ServiceCallback, RequestOptionsBase, Serializer, OperationSpec } from "@azure/core-http";
```
Comment (ceciliaavila) [inline libraries/botframework-connector/src/tokenApi/operations/userToken.ts:6]: Use single quotes here.

File: libraries/tests.uischema [hunk 4]
```diff
@@ -8,7 +8,8 @@
         "triggers",
         "generator",
         "selector",
-        "schema"
+        "schema",
+        "dialogs"
```
Comment (ceciliaavila) [inline libraries/tests.uischema:12]: What's this change about?

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

File: libraries/tests.uischema (hunk 4 above)
Comment (JhontSouth) [reply libraries/tests.uischema:12]: to avoid errors with test:schema

File: package.json [hunk 5]
```diff
@@ -54,11 +54,11 @@
     "**/request-promise/tough-cookie": "^4.1.3",
     "**/request-promise-native/tough-cookie": "^4.1.3",
     "**/istanbul-lib-instrument/@babel/core": "^7.23.2",
-    "**/@babel/plugin-proposal-class-properties/@babel/core": "^7.23.2"
+    "**/@babel/plugin-proposal-class-properties/@babel/core": "^7.23.2",
+    "@azure/core-util": "1.5.0"
```
Comment (sw-joelmut) [inline package.json:58]: Same here

**Review by sw-joelmut — CHANGES_REQUESTED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Comment by JhontSouth** [comment]
Promoted /4578

**Pull request closed without merging**